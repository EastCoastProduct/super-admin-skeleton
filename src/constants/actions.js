export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const LOGOUT_SUCCESS = 'LOGOUT_SUCCESS';

export const USERS_GET_FAILED = 'USERS_GET_FAILED';
export const USERS_GET_SUCCESS = 'USERS_GET_SUCCESS';
export const USERS_PAGINATION_CHANGE = 'USERS_PAGINATION_CHANGE';

export const USER_GET_FAILED = 'USER_GET_FAILED';
export const USER_FETCH_SUCCESS = 'USER_FETCH_SUCCESS';
