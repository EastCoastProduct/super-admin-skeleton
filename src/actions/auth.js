import store from 'store';
import { LOGIN_SUCCESS, LOGOUT_SUCCESS } from '../constants/actions';
import { API_URL } from '../constants/application';
import parseErrors from '../utils/parseErrors';
import fetch from '../utils/fetch';

export const loginSuccess = superadmin => ({
  type: LOGIN_SUCCESS,
  superadmin,
});

export const logoutSuccess = () => ({
  type: LOGOUT_SUCCESS,
});

export const loginFetch = (values, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/authenticate`, {
      method: 'POST',
      body: JSON.stringify(values),
    }).then((resp) => {
      store.set('token', `Bearer ${resp.token}`);
      store.set('superadmin', resp.user);
      dispatch(loginSuccess(resp.user));
      return typeof cb === 'function' && cb();
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );

export const logoutAction = cb =>
  (dispatch) => {
    store.clear();
    dispatch(logoutSuccess());
    return typeof cb === 'function' && cb();
  };
