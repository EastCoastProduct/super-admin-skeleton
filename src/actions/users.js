import queryString from 'query-string';
import { USERS_GET_SUCCESS, USERS_GET_FAILED, USERS_PAGINATION_CHANGE }
  from '../constants/actions';
import { API_URL } from '../constants/application';
import fetch from '../utils/fetch';

export const usersGetSuccess = (list, listTotal) => ({
  type: USERS_GET_SUCCESS,
  list,
  listTotal,
});

export const usersGetFailed = error => ({
  type: USERS_GET_FAILED,
  error,
});

export const paginationChange = page => ({
  type: USERS_PAGINATION_CHANGE,
  page,
});

export const usersGetFetch = (qs, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/users?${queryString.stringify(qs)}`)
      .then((resp) => {
        dispatch(usersGetSuccess(resp.rows, resp.count));
        return typeof cb === 'function' && cb();
      }).catch(err =>
        dispatch(usersGetFailed(err.message))
      );
