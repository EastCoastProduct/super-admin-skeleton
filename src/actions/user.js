import { USER_FETCH_SUCCESS, USER_GET_FAILED } from '../constants/actions';
import { API_URL } from '../constants/application';
import createFormData from '../utils/createFormData';
import parseErrors from '../utils/parseErrors';
import fetch from '../utils/fetch';

export const userFetchSuccess = profile => ({
  type: USER_FETCH_SUCCESS,
  profile,
});

export const userGetFailed = error => ({
  type: USER_GET_FAILED,
  error,
});

export const userGetFetch = userId =>
  dispatch =>
    fetch(`${API_URL}/users/${userId}`).then(resp =>
      dispatch(userFetchSuccess(resp))
    ).catch(err =>
      dispatch(userGetFailed(err.message))
    );

export const userUpdateFetch = (values, userId, cb) =>
  dispatch =>
    fetch(`${API_URL}/users/${userId}`, {
      method: 'POST',
      body: createFormData(values),
    }).then((resp) => {
      dispatch(userFetchSuccess(resp));
      return typeof cb === 'function' && cb();
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );

export const userCreateFetch = (values, cb) =>
  dispatch =>
    fetch(`${API_URL}/superAdmin/users`, {
      method: 'POST',
      body: JSON.stringify(values),
    }).then((resp) => {
      dispatch(userFetchSuccess(resp));
      return typeof cb === 'function' && cb(resp.id);
    }).catch(err =>
      Promise.reject(parseErrors(err))
    );
