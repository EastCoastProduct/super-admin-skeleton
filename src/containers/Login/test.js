import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';
import deepMerge from 'deepmerge';
import { LoginComponent } from './';
import * as Actions from '../../actions/auth';

jest.mock('../../images/ecp_logo.png', () => 'ecp_logo.png');

describe('Login component', () => {
  const props = {
    dispatch: jest.fn(() => Promise.resolve()),
    form: 'Login',
    handleSubmit: f => f,
    router: {
      push: jest.fn(),
    },
    submitting: true,
  };
  const wrapper = shallow(<LoginComponent {...props} />);

  describe('snapshot', () => {
    it('renders required data', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('renders required data with error', () => {
      const newProps = deepMerge(props, {
        error: 'User not found.',
        submitting: false,
      });
      const newWrapper = shallow(<LoginComponent {...newProps} />)

      expect(newWrapper).toMatchSnapshot();
    });
  });

  describe('instance', () => {
    afterEach(() => {
      jest.resetAllMocks(); // move to before when clearAllMocks gets out
    });
    Actions.loginFetch = jest.fn();
    const instance = wrapper.instance();

    it('handleLogin method', () => {
      const values = fromJS({
        email: 'test@email.com',
        password: 'Aa123456',
      });
      const p = instance.handleLogin(values);

      expect(Actions.loginFetch).toHaveBeenCalledWith(values);
      expect(instance.props.dispatch).toHaveBeenCalled();
      return p.then(() => {
        expect(instance.props.router.push).toHaveBeenCalledWith('/');
      });
    });
  });
});
