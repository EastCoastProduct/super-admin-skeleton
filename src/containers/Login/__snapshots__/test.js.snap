exports[`Login component snapshot renders required data 1`] = `
<main
  className="page_ftdrhl">
  <section
    className="box_1306j1v">
    <img
      alt="Company Logo"
      className="logo_2al8b9"
      src="ecp_logo.png" />
    <h1
      className="header_ky8f4n">
      Login to your account
    </h1>
    <form
      noValidate={true}
      onSubmit={[Function]}>
      <Field
        component={[Function]}
        id="Login"
        label="Email"
        name="email"
        placeholder="Email"
        type="email"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
        validated={true} />
      <Field
        component={[Function]}
        id="Login"
        label="Password"
        name="password"
        placeholder="Password"
        type="password"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
        validated={true} />
      <Button
        className={
          Object {
            "_definition": Object {
              "width": "39%",
            },
            "_name": "button_ltlwnu",
          }
        }
        disabled={true}>
        Login
      </Button>
    </form>
  </section>
</main>
`;

exports[`Login component snapshot renders required data with error 1`] = `
<main
  className="page_ftdrhl">
  <section
    className="box_1306j1v-o_O-boxErr_wwds3j">
    <img
      alt="Company Logo"
      className="logo_2al8b9"
      src="ecp_logo.png" />
    <h1
      className="header_ky8f4n">
      Login to your account
    </h1>
    <form
      noValidate={true}
      onSubmit={[Function]}>
      <Field
        component={[Function]}
        id="Login"
        label="Email"
        name="email"
        placeholder="Email"
        type="email"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
        validated={true} />
      <Field
        component={[Function]}
        id="Login"
        label="Password"
        name="password"
        placeholder="Password"
        type="password"
        validate={
          Array [
            [Function],
            [Function],
          ]
        }
        validated={true} />
      <Button
        className={
          Object {
            "_definition": Object {
              "width": "39%",
            },
            "_name": "button_ltlwnu",
          }
        }
        disabled={false}>
        Login
      </Button>
      <ErrorMsg
        className={
          Object {
            "_definition": Object {
              "margin": "10px 0 0",
            },
            "_name": "error_1f65vod",
          }
        }>
        User not found.
      </ErrorMsg>
    </form>
  </section>
</main>
`;
